### PR TITLE
Search: pull out search interface

### PR DESCRIPTION
**Pull request by rvantonder**
Use dedicated types to improve readability. Also deleted a comment about `search2.go`

Test plan: Tests pass.

### CODE REVIEW

File: cmd/frontend/graphqlbackend/search.go [hunk 1]
```diff
@@ -55,19 +51,23 @@ func maxReposToSearch() int {
...
+type searchArgs struct {
 	Version     string
 	PatternType *string
 	Query       string
 	After       *graphql.ID
 	First       *int32
-}) (interface {
+}
+
+type searchIntf interface {
```
Comment (unknwon) [inline cmd/frontend/graphqlbackend/search.go]: Unless there is a naming conflict, I think this is more Go idiomatic:

```suggestion
type searchResolver interface {
```

Comment (rvantonder) [reply cmd/frontend/graphqlbackend/search.go]: Can't: https://sourcegraph.com/github.com/sourcegraph/sourcegraph@4302f2bd8542ecf733ec8d37d31e4b80e9c21c90/-/blob/cmd/frontend/graphqlbackend/search.go#L158

Comment (rvantonder) [reply cmd/frontend/graphqlbackend/search.go]: Though if you have an alternative idiomatic suggestion, LMK.

Comment (unknwon) [reply cmd/frontend/graphqlbackend/search.go]: How's `searchResponser`?

Comment (mrnugget) [reply cmd/frontend/graphqlbackend/search.go]: Or `searchResulter`? `searchIntf` is certainly not Go-ey

Comment (rvantonder) [reply cmd/frontend/graphqlbackend/search.go]: See #6170.

**Pull request merged**